### ISSUE TITLE
[Paper Note] The Impact of Initialization on LoRA Finetuning Dynamics, Soufiane Hayou+, NeurIPS'24, 2024.06

### ISSUE THREAD

**Issue by AkihikoWatanabe** [AkihikoWatanabe/paper_notes#2978]
# URL
- https://arxiv.org/abs/2406.08447
# Authors
  - Soufiane Hayou
  - Nikhil Ghosh
  - Bin Yu
# Abstract
  - In this paper, we study the role of initialization in Low Rank Adaptation (LoRA) as originally introduced in Hu et al. (2021). Essentially, to start from the pretrained model as initialization for finetuning, one can either initialize B to zero and A to random (default initialization in PEFT package), or vice-versa. In both cases, the product BA is equal to zero at initialization, which makes finetuning starts from the pretrained model. These two initialization schemes are seemingly similar. They should in-principle yield the same performance and share the same optimal learning rate. We demonstrate that this is an incorrect intuition and that the first scheme (initializing B to zero and A to random) on average yields better performance compared to the other scheme. Our theoretical analysis shows that the reason behind this might be that the first initialization allows the use of larger learning rates (without causing output instability) compared to the second initialization, resulting in more efficient learning of the first scheme. We validate our results with extensive experiments on LLMs.
# Translation (by gpt-4o-mini)
- 本論文では、Hu et al. (2021)で最初に提案されたLow Rank Adaptation (LoRA)における初期化の役割を研究します。基本的に、ファインチューニングのための初期化として事前学習済みモデルから始めるには、Bをゼロに初期化しAをランダムに初期化する（PEFTパッケージのデフォルト初期化）か、その逆を行うことができます。いずれの場合も、初期化時の積BAはゼロとなり、ファインチューニングは事前学習済みモデルから始まります。これらの2つの初期化方式は一見似ているように見えます。原則として、同じパフォーマンスをもたらし、同じ最適学習率を共有するはずです。しかし、これは誤った直感であり、最初の方式（Bをゼロに初期化しAをランダムに初期化する）が平均して他の方式よりも優れたパフォーマンスを発揮することを示します。我々の理論的分析は、最初の初期化が2番目の初期化と比較してより大きな学習率を使用できること（出力の不安定性を引き起こさずに）を可能にし、これが最初の方式のより効率的な学習につながる可能性があることを示しています。LLMsに関する広範な実験を通じて、我々の結果を検証します。
# Summary (by gpt-4o-mini)
- 本論文では、LoRAにおける初期化の役割を研究し、Bをゼロに初期化しAをランダムに初期化する方式が他の方式よりも優れたパフォーマンスを示すことを明らかにします。この初期化方式は、より大きな学習率を使用できるため、効率的な学習を促進する可能性があります。LLMsに関する実験を通じて結果を検証します。

**Comment by AkihikoWatanabe** [comment]
元ポスト:https://x.com/jxmnop/status/1970893830619894186?s=46&t=Y6UuIHB0Lv0IpmFAjlc2-Q

**Comment by AkihikoWatanabe** [comment]
初期化でBをzeroにするという手法は以下でも提案されているが、本研究の方が下記研究よりも投稿が1年程度早い:
- #2194

**Comment by AkihikoWatanabe** [comment]
openreview:https://openreview.net/forum?id=sn3UrYRItk&referrer=%5Bthe%20profile%20of%20Nikhil%20Ghosh%5D(%2Fprofile%3Fid%3D~Nikhil_Ghosh1)
